### DISCUSSION TITLE
Contributor call: 5 August 2023

### DISCUSSION THREAD

**Discussion by l3xander** [freesewing/freesewing#4622] [Support]
You are all invited to the FreeSewing Contributor Call.
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This page will be updated after the call with notes.

### About
FreeSewing's contributor call is an informal hangout/chat/meeting that is open to all FreeSewing contributors and users alike.
The call is organized/hosted by @tangerineshark.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

**Details**
Location: https://discord.freesewing.org (meeting room: 🔊`voice chat`)
Date: August 5th, 2023 4pm UTC, which can be found in your local time zone [here](https://savvytime.com/converter/utc-to-belgium-antwerpen-ca-los-angeles-il-chicago-ny-new-york-city-brazil-sao-paulo-united-kingdom-london-france-paris-south-africa-johannesburg-msd-india-mumbai-hkt-japan-tokyo-australia-sydney/aug-5-2023/4pm).
Or check the Events tab in Discord!
The meetings will always be at 6pm, Antwerp time.

### Agenda
- Joost's Discord changes
- IRL contributing-to-FreeSewing night
- Orphaned patterns

**Comment by joostdecock** [comment]
In case you need inspiration for the agenda:

- **Discord changes**: In anticipation/preparation of the v3 release, I'd like to make some changes to the way Discord is organized. Specifically:
    - One **Support** channel where users can turn for help. Replaces the various help-* channels because if you want help you should not have to worry too much about where to ask for it. Discord now supports threads, so it's not that hard for people to start up a thread in case two help topics are getting mixed.
    - The **info** channel can probably use a refresh (if we opt to keep it, content there is 18 months old)
    - I would merge **pattern-showcase** and **other-showcase** into a single **showcase** channel because once again, I don't think people need to overthink it. We don't mind if people post something else, but having **other-showcase** seems to actively encourage it and I'm not sure that's what we want
    - The **any-questions** channel needs to go IMHO. Discord is only for ephemeral content. It is not a good idea/strategy to try to build a knowledge base there because at the end of the day, it's still a walled garden. Instead, questions that come up regularly should be added to the FAQ on the website.

**Comment by l3xander** [comment]
Thank you @joostdecock for the ideas. I agree the Discord could use some changing.
First of, `any-questions` is not intended to be used as a knowledge base. It's intended to be a better version of threads. But I'll get to that. 

### The Discord
I went ahead and made notes for this part of the meeting as the discord is my responsibility (lovingly).

**On help channels**
Thoughts:
- Perhaps the `#development-help` channel can go to the `contributor teams` folder. (@bobgeorgethe3rd)
  We then would have to change names here and there, as we'd have both `#development-help`, which is more for people who are making their own FreeSewing pattern, and `#development`, which is more for FreeSewing infrastructure. (@eriese)
- General agreement in the call that the forum channel (previously `any-questions`) is not used much and therefore not useful in its current state.
- A `#freesewing-help` channel is also a possibility. (@bobgeorgethe3rd)
- Perhaps for pattern making help we want to make a distinction between a sewing perspective (e.g. "What kind of curve would work well for the armhole of this pattern?") and a programming perspective (e.g. "How do I program this type of curve?"). (@/Thrunic & @bobgeorgethe3rd) (Doesn't have to be multiple channels for this.)
- @joostdecock prefers fewer channels, however in call there was a general disagreement with that. People in call prefer for there to be multiple channels with each a clear intention. 
- For a server with enough questions that it happens that people sort of interrupt each other in one channel (or are afraid to ask a question because another question is in the process of being answered in a channel), I think forum channels (previously `any-questions`) are the better solution over threads, because forum channels allow for tagging and organization, and threads show up as just a list. I also think for people new to discord it’s easier to make a forum post than a thread. Though I do think we should keep in mind the barrier that can exist when you expect people to _make_ a post (can scare people off). (@lexander2002)
- We can't/shouldn't expect people to look back in the forum channel to see if a question has already been asked (nobody really does that anyways). 
- If we do the forum feature, we shouldn't have both a forum channel and the help channels, because then it becomes unclear where to ask a question. Also a note that people are (currently) more active in the normal channels than the forum channel, because they can just check the channel of their 'expertise'. (@SeaZeeZee & @eriese) Tags in forum posts can help with this.
- We can move the forum channel up, rename it to something clearer (it's now `need-help`), and rename the `-help` channels to `-discussion` channels. We keep an eye on how that goes and change it if it doesn't work. (@bobgeorgethe3rd) 
  General agreement in call to try this out.
- We should pin the "I didn't receive an email"-question as that gets asked a LOT. (@eriese) 
  In general we should have the FAQ pinned so people have easy access to it from the help channel. (@lexander2002) But not too many pinned posts, it's supposed to stay light.
- We also decided it's best to archive `#website-help`.
I want to add we should look out that `-discussion` channels don't become too vague.

**On showcase channels**
Thoughts:
- If we're archiving `#other-showcase`, people can post in `#pics-pets-memes` instead. (@lexander2002)
- It would be nice for a non-FreeSewing showcase channel to exist because it builds community. (@bobgeorgethe3rd & @SeaZeeZee)
- We can change `#pattern-showcase` to `#freesewing-showcase` for extra clarity. (@bobgeorgethe3rd)
- It's rare for `#pattern-showcase` to have a non-FreeSewing post in it, and those who really only want FreeSewing-only showcases will go to the site. Maybe we can have a `#sewing-showcase` instead of `#other-showcase` so we can keep the discussions that exist now. (@eriese)
- We can also change `#other-showcase` to `#brag` or `#humble-brag`. (@SeaZeeZee) That would be a clear distinction.
- Somewhat related side-note, there are a number of people in the server who don't actually know about the FreeSewing site. (@SeaZeeZee) 
  The site should be added to the onboarding feature (the screen people see when they first join the server). (@bobgeorgethe3rd) It will need updating after channel changes anyways. 

**On the info channel**
We figured this doesn't need much discussion in the call. We made the decision to wait with updating the `#info` channel until after V3 is released to avoid broken links, and then we also avoid linking to (help) channels that we may decide to change soon (as the current layout - forum channel + discussion channels - is in a test phase). (@tangerineshark & @lexander2002)

**Todos**
(Thank you to those who have already done some of these.)
- [x] Move `any-questions` up and rename it (now `need-help`).
- [x] Change help channels to discussion channels
- [x] Archive `#website-help`
- [x] Pin the mail question and the FAQ
- [x] Update server onboarding, include website
- [x] Change showcase channels

### IRL FreeSewing
I also made some notes on the second agenda point. "De War" in Amersfoort, The Netherlands, has been hosting FreeSewing nights (three so far), where people get to know FreeSewing and its features. De War is doing a conference next week where there will be another FreeSewing evening on the 10th, but this time it's more of a hackathon, and we are going to try out the bib tutorial and in general look at ways on how to contribute to FreeSewing.
There's excitement from De War to work with V3 already, but from @SeaZeeZee and @eriese (and @bobgeorgethe3rd) it's clear that the current V3 interface is not so loved, so @eriese is going to try to figure out a way for us to work with V3 code but the V2.5 interface.
It was also mentioned to use Gitpod for Windows users if they don't already have a WSL set up. (But Linux seems to be mainly used there.) 
(Thanks for the help)

**Reply by l3xander** [reply]
Other planned changes for the server: integrating the server better with the newer features Discord has brought out. (Already enabled them but not fully.) It has more intuitive features now which make `#customize-profile` and `#rules` obsolete. Also the `#meetings` channel is a bit....... yeah. I need to think about that. I also think we should reconsider the media channels, they don't serve much of a purpose now. Not enough changes happen in the Discord for it to need its own channel, although Discord the application does need a channel somewhere to post its Admin updates to. And the social media are not active so there are no posts cross-posted to the Discord, I should also relook at the bots I used for that. 
🥶

**Reply by eriese** [reply]
Some clarification on what I think will work for the hackathon:

The current version at `npx @freesewing/new-design@next` is working, and runs v3 code on lab2.5, so unless @joostdecock is comfortable with us publishing an `@edge ` version using the current state of the ui, that's the version to use. It would be work to get `new-design` ready for that, but potentially doable in the time.

For Windows users @lexander2002 I recommend that ahead of the event you run `npx @freesewing/new-design@next` locally, and then push the result to a repo in your own github. People can then clone that repo to their local systems and run it with no issues. @SeaZeeZee  has tested this. It doesn't have the same pizzazz as running a node script, but it's a completely legit way for people to get a codebase.

The docs for the tutorial are up to date, but could use some tweaking for clarity, and need images that reflect lab2.5. @SeaZeeZee has started work on the tweaks, and it'd be helpful if someone could do the screen shots

**Reply by l3xander** [reply]
Oh that sounds pretty smart, I'll try to get that working. Thank you!
